### PR TITLE
tests/server/api: make `TestPreparingProgress` stable

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: Ref #7969

### What is changed and how does it work?

because `checkStores` checks store state every 3 seconds

https://github.com/tikv/pd/blob/227860912dbf95c19c506335be2e6f8319ae28ba/server/cluster/cluster.go#L1796-L1804

when `GetTotalRegionCount` is not meet `InitClusterRegionThreshold`(100), will change store 4 to the serving state

https://github.com/tikv/pd/blob/6ecf23def91927788d6fb5103368c1755fe0dded/tests/server/api/api_test.go#L957-L963

and when changing store 4 to the serving state will make test unstable

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/server/api/api_test.go [hunk 1]
```diff
@@ -969,13 +969,20 @@ func TestPreparingProgress(t *testing.T) {
 			StartTimestamp: time.Now().UnixNano() - 100,
 		},
 	}
-
-	for _, store := range stores {
+	// store 4 and store 5 are preparing state while store 1, store 2 and store 3 are state serving state
+	for _, store := range stores[:2] {
```
Comment (lhy1024) [inline tests/server/api/api_test.go:973]: Don't we need to add store1,2,3 which never seem to be added?

Comment (HuSharp) [reply tests/server/api/api_test.go:973]: stores[:2] is for 1/2/3 and stores[2:] is for 4/5

File: tests/server/api/api_test.go [hunk 2]
```diff
@@ -969,13 +969,20 @@ func TestPreparingProgress(t *testing.T) {
...
 	}
-	for i := 0; i < 100; i++ {
+	for i := 0; i < core.InitClusterRegionThreshold; i++ {
 		tests.MustPutRegion(re, cluster, uint64(i+1), uint64(i)%3+1, []byte(fmt.Sprintf("%20d", i)), []byte(fmt.Sprintf("%20d", i+1)), core.SetApproximateSize(10))
 	}
+	testutil.Eventually(re, func() bool {
+		return leader.GetRaftCluster().GetTotalRegionCount() == core.InitClusterRegionThreshold
+	})
+	// to avoid forcing the store to the `serving` state with too few regions
+	for _, store := range stores[2:] {
```
Comment (HuSharp) [inline tests/server/api/api_test.go:983]: @lhy1024 PTAL :)

**Review by lhy1024 — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**